tracee-rules/input.go

```diff
@@ -66,8 +68,132 @@ func setupTraceeSource(traceeFilePath string) (chan types.Event, error) {
 			}
 		}
-		f.Close()
+		opts.inputFile.Close()
+		close(res)
+	}()
+	return res, nil
+}
+
+func setupJSONInputSource(opts *traceeInputOptions) (chan types.Event, error) {
+	res := make(chan types.Event)
+	scanner := bufio.NewScanner(opts.inputFile)
+	go func() {
+		for scanner.Scan() {
+			event := scanner.Bytes()
+			var e tracee.Event
+			err := json.Unmarshal(event, &e)
+			if err != nil {
+				log.Printf("invalid json in %s: %v", string(event), err)
+			}
+			res <- tracee.Event(e)
+		}
+		opts.inputFile.Close()
 		close(res)
 	}()
 	return res, nil
 }
+
+func parseTraceeInputOptions(inputOptions []string) (*traceeInputOptions, error) {
+
+	var (
+		inputSourceOptions traceeInputOptions
+		err                error
+	)
+
+	if len(inputOptions) == 0 {
+		return nil, errors.New("no tracee input options specified")
+	}
+
+	for i := range inputOptions {
+		if inputOptions[i] == "help" {
+			return nil, helpErr
+		}
+
+		kv := strings.Split(inputOptions[i], ":")
+		if len(kv) != 2 {
+			return nil, fmt.Errorf("invalid input-tracee option: %s", inputOptions[i])
+		}
+		if kv[0] == "" || kv[1] == "" {
+			return nil, fmt.Errorf("empty key or value passed: key: >%s< value: >%s<", kv[0], kv[1])
+		}
+		if kv[0] == "file" {
+			err = parseTraceeInputFile(&inputSourceOptions, kv[1])
+			if err != nil {
+				return nil, err
+			}
+		} else if kv[0] == "option" {
+			err = parseTraceeInputOption(&inputSourceOptions, kv[1])
+			if err != nil {
+				return nil, err
+			}
+		} else if kv[0] == "format" {
+			err = parseTraceeInputFormat(&inputSourceOptions, kv[1])
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			return nil, fmt.Errorf("invalid input-tracee option key: %s", kv[0])
+		}
+	}
+	return &inputSourceOptions, nil
+}
+
+func parseTraceeInputFile(option *traceeInputOptions, fileOpt string) error {
+
+	if fileOpt == "stdin" {
+		option.inputFile = os.Stdin
+		return nil
+	}
+	_, err := os.Stat(fileOpt)
+	if err != nil {
+		return fmt.Errorf("invalid Tracee input file: %s", fileOpt)
+	}
+	f, err := os.Open(fileOpt)
+	if err != nil {
+		return fmt.Errorf("invalid file: %s", fileOpt)
+	}
+	option.inputFile = f
+	return nil
+}
+
+func parseTraceeInputOption(option *traceeInputOptions, optionString string) error {
+	if strings.ToUpper(optionString) == "EOF" {
+		option.closeOnEOF = true
+	} else {
+		return fmt.Errorf("unkown tracee input option: %s", optionString)
+	}
+	return nil
+}
+
+func parseTraceeInputFormat(option *traceeInputOptions, formatString string) error {
+	formatString = strings.ToUpper(formatString)
+
+	if formatString == "JSON" {
+		option.inputFormat = jsonInputFormat
+	} else if formatString == "GOB" {
+		option.inputFormat = gobInputFormat
+	} else {
+		option.inputFormat = invalidInputFormat
+		return fmt.Errorf("invalid tracee input format specified: %s", formatString)
+	}
+	return nil
+}
+
+func printHelp() {
+	traceeInputHelp := `
+tracee-rules --input-tracee <key:value>,<key:value> --input-tracee <key:value>
```

Comment:
To be future proof, I think we should change this flag name to be simply `input`.
Then, we can have something like:
`--input source:key=value`
WDYT?

Reply:
It's unlikely that future input sources will share config options. consider kubernetes audit - it needs a URL, credentials, maybe a kubeconfig file. none of this is relevant to tracee, so I think it's better to keep separate config structs per input source and separate flags

Reply:
_(see my comment above)_